### PR TITLE
feat: pass anonymousId to language server [ROAD-1185]

### PR DESCRIPTION
**Pull request by michelkaporin**
### Description

Ensures `anonymousId` synchronisation for analytics.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### CODE REVIEW

File: src/snyk/common/languageServer/languageServer.ts [hunk 1]
```diff
@@ -126,6 +125,7 @@ export class LanguageServer implements ILanguageServer {
       ...settings,
       integrationName: CLI_INTEGRATION_NAME,
       integrationVersion: await Configuration.getVersion(),
+      deviceId: this.user.anonymousId,
```
Comment (bastiandoetsch) [inline src/snyk/common/languageServer/languageServer.ts:128]: why are these 4 fields not part of the settings object?

File: src/test/unit/common/languageServer/languageServer.test.ts [hunk 2]
```diff
@@ -2,26 +2,25 @@
 import assert, { deepStrictEqual } from 'assert';
 import { ReplaySubject } from 'rxjs';
 import sinon from 'sinon';
+import { v4 } from 'uuid';
```
Comment (bastiandoetsch) [inline src/test/unit/common/languageServer/languageServer.test.ts:5]: why not v1?

File: src/test/unit/common/languageServer/languageServer.test.ts [hunk 3]
```diff
@@ -2,26 +2,25 @@
...
 import { IVSCodeWorkspace } from '../../../../snyk/common/vscode/workspace';
-import { extensionContextMock } from '../../mocks/extensionContext.mock';
 import { LoggerMock } from '../../mocks/logger.mock';
 import { windowMock } from '../../mocks/window.mock';
 import { stubWorkspaceConfiguration } from '../../mocks/workspace.mock';
 
 suite('Language Server', () => {
-  const extensionContext = extensionContextMock;
   const authService = {} as IAuthenticationService;
+  const user = new User(v4(), undefined);
```
Comment (bastiandoetsch) [inline src/test/unit/common/languageServer/languageServer.test.ts:21]: why not v1?

**Review by bastiandoetsch — APPROVED** [review]

File: src/test/unit/common/languageServer/languageServer.test.ts (hunk 3 above)
Comment (michelkaporin) [reply src/test/unit/common/languageServer/languageServer.test.ts:21]: current logic used v4 for that. But it doesn't matter as both are arbitrary.

File: src/snyk/common/languageServer/languageServer.ts (hunk 1 above)
Comment (michelkaporin) [reply src/snyk/common/languageServer/languageServer.ts:130]: added an explanation comment 👍

**Pull request merged**